src/test/java/picard/sam/markduplicates/AbstractMarkDuplicatesCommandLineProgramTester.java

```diff
@@ -143,33 +144,33 @@ public void setExpectedOpticalDuplicate(final int expectedOpticalDuplicatePairs)
     }
 
     @Override
-    public void test() {
+    public void test() throws IOException {
         try {
             updateExpectedDuplicationMetrics();
 
             // Read the output and check the duplicate flag
             int outputRecords = 0;
             final Set<String> sequencingDTErrorsSeen = new HashSet<>();
-            final SamReader reader = SamReaderFactory.makeDefault().open(getOutput());
-            for (final SAMRecord record : reader) {
-                outputRecords++;
-                final String key = samRecordToDuplicatesFlagsKey(record);
-                if (!this.duplicateFlags.containsKey(key)) {
-                    System.err.println("DOES NOT CONTAIN KEY: " + key);
-                }
-                Assert.assertTrue(this.duplicateFlags.containsKey(key));
-                final boolean value = this.duplicateFlags.get(key);
-                this.duplicateFlags.remove(key);
-                if (value != record.getDuplicateReadFlag()) {
-                    System.err.println("Mismatching read:");
-                    System.err.print(record.getSAMString());
-                }
-                Assert.assertEquals(record.getDuplicateReadFlag(), value);
-                if (testOpticalDuplicateDTTag && MarkDuplicates.DUPLICATE_TYPE_SEQUENCING.equals(record.getAttribute("DT"))) {
-                    sequencingDTErrorsSeen.add(record.getReadName());
+            try(final SamReader reader = SamReaderFactory.makeDefault().open(getOutput())) {
+                for (final SAMRecord record : reader) {
+                    outputRecords++;
+                    final String key = samRecordToDuplicatesFlagsKey(record);
+                    if (!this.duplicateFlags.containsKey(key)) {
+                        System.err.println("DOES NOT CONTAIN KEY: " + key);
+                    }
+                    Assert.assertTrue(this.duplicateFlags.containsKey(key));
+                    final boolean value = this.duplicateFlags.get(key);
+                    this.duplicateFlags.remove(key);
+                    if (value != record.getDuplicateReadFlag()) {
+                        System.err.println("Mismatching read:");
```

Comment:
Also redundant with assert on line 168 below. If showing `record.getSAMString()` is helpful for debugging this error, add it to the `assertEquals()` call as part of a message argument. e.g.
```
Assert.assertEquals(record.getDuplicateReadFlag(), value, "Mismatching read: " + record.getSAMString());
```